main.py

```diff
@@ -300,23 +270,38 @@ def __init__(
         self.directory_result_map: dict[str, dict] = {}
         self.file_result_map: dict[str, str] = {}
         self.total_count = 0
-        self.progress = 0
         self.duration = datetime.timedelta()
+        self.parse_only = parse_only
+        self.update_function = nop
+        self.print_output = print
+        self.lists = lists
+        self.shuffle = shuffle
 
     def log(self, message: str) -> None:
         if not self.silent and not self.per_file:
-            print(message)
+            self.print_output(message)
 
-    def find_tests(self, pattern: str) -> None:
-        self.log("Searching test files...")
+    def find_tests(self, pattern: str, ignore: str) -> None:
         if Path(pattern).resolve().is_file():
             self.files = [Path(pattern).resolve()]
         else:
-            self.files = [
-                path.resolve()
-                for path in self.test262_root.glob(pattern)
-                if path.is_file() and not path.stem.endswith("FIXTURE")
-            ]
+            ignored_files = glob.glob(
+                os.path.join(self.test262_root, ignore), recursive=True
+            )
+            for p in glob.iglob(
+                os.path.join(self.test262_root, pattern), recursive=True
+            ):
```

Comment:
Please don't use `os.path` in any modern Python code, use pathlib instead.

Reply:
I now use Pathlib to resolve the path.
Is `glob.glob` fine? I use that instead of pathlib glob since it allows for `/**/` to also mean zero directories which I sometimes wanted (with `recursive=True`). I can revert this to pathlib if that is better.